p2p/net/swarm/swarm_listen.go

```diff
@@ -39,7 +39,7 @@ func (s *Swarm) Listen(addrs ...ma.Multiaddr) error {
 
 // ListenClose stop and delete listeners for all of the given addresses.
 func (s *Swarm) ListenClose(addrs ...ma.Multiaddr) {
-	var listenersToClose []transport.Listener
+	listenersToClose := make([]transport.Listener, 0, len(s.listeners.m))
```

Comment:
This will massively over-allocate.

Reply:
I see. I don't find any good ways to provide a size hint here, so removing this